cmd/lookoutd/common.go

```diff
@@ -375,8 +379,10 @@ func (c *queueConsumerCommand) startServer(srv *lookout.DataServerHandler) error
 	go func() {
 		if err := grpcSrv.Serve(lis); err != nil {
 			log.Errorf(err, "data server failed")
+			os.Exit(1)
```

Comment:
Why is this needed now? Are there any changes that require it, or was it missing from before?

Reply:
it was missed before I just noticed and fixed it while doing my task.

Currently, if data-server fails we log it but don't exit.
Worker will continue processing events but all of them will fail because analyzers won't be able to connect to data-server to get changes.

Reply:
How did you stumble with the problem? Can this happen only on initialization, like an already used port, or at any time?

Reply:
I just noticed it while looking at the code around.
I didn't dig too deep in grpc implementation to answer your question. But for http server it can return error in 2 cases on initialization and when done channel is closed. (happens when you call `server.Shutdown()` or `server.Close()`) Most probably grpc works similar.

Reply:
What I was thinking is that maybe this method should be blocking, removing `go func` and directly do `return grpcSrv.Serve`. Then on top handle it like here https://github.com/src-d/lookout/blob/master/cmd/lookoutd/serve.go#L96. WDYT? is it worth to make it consistent with other top level errors, so we have one clear exit point?

Reply:
We use this function not only in `serve.go` but in worker too. It doesn't use chan/switch pattern. SDK uses a similar pattern but instead of making the methods blocking it accepts chan as an argument. 

Though I like your idea to make it consistent, I'm not sure which pattern we should use. Do you mind if we move it to another issue then?

Reply:
ok, let's keep it as it is here, and I'll open an issue to look into it later, it's a minor thing and not really urgent.

Reply:
Here: https://github.com/src-d/lookout/issues/362